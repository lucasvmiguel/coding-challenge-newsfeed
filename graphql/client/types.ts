export enum resolveTypes {
  User = 'User',
  Announcement = 'Announcement',
  Project = 'Project'
}
