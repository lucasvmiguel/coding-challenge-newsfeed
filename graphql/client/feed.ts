import { gql } from '@apollo/client'

export const FEED_QUERY = gql`
  query Feed($user_type: Fellowship!, $limit: Int, $offset: Int) {
    feed(user_type: $user_type, limit: $limit, offset: $offset) {
      __typename
      ... on User {
        id
        name
        bio
        avatar_url
        fellowship
        created_ts
        projects {
          id
          name
          icon_url
        }
      }
      ... on Announcement {
        id
        title
        body
        fellowship
        created_ts
      }
      ... on Project {
        id
        name
        description
        icon_url
        created_ts
        users {
          id
          name
          avatar_url
        }
      }
    }
  }
`
