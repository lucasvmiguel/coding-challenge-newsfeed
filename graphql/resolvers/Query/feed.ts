import db, { UserRow, AnnouncementRow, ProjectRow, Fellowship } from '../../db'
import { limitQueryParam, offsetQueryParam } from '../../pagination'

type Args = {
  user_type: Fellowship
  limit: number;
  offset: number;
}

type FeedItem = UserRow | AnnouncementRow | ProjectRow

export default async function feed(parent: unknown, { user_type, limit, offset }: Args): Promise<FeedItem[]> {
  const limitQP = limitQueryParam(limit)
  const offsetQP = offsetQueryParam(offset)

  switch (user_type) {
    case Fellowship.Writers:
      return writersQuery(limitQP, offsetQP)
    case Fellowship.Founders:
      return foundersQuery(limitQP, offsetQP)
    case Fellowship.Angels:
      return angelsQuery(limitQP, offsetQP)
    default:
      return []
  }
}

const writersQuery = (limitQP: number, offsetQP: number) => {
  return db.getAll(`
      SELECT a.id, a.title, NULL as name, a.body, NULL as bio, NULL as description, a.created_ts, a.fellowship, NULL as avatar_url, NULL as icon_url FROM announcements a
      WHERE a.fellowship == "writers" OR a.fellowship == "all"
      UNION ALL
      SELECT u.id, NULL as title, u.name, NULL as body, u.bio, NULL as description, u.created_ts, u.fellowship, u.avatar_url, NULL as icon_url FROM users u
      WHERE u.fellowship == "writers"
      ORDER BY created_ts DESC
      LIMIT ? OFFSET ?
    `,
    [limitQP, offsetQP]
  )
}

const foundersQuery = (limitQP: number, offsetQP: number) => {
  return db.getAll(`
      SELECT a.id, a.title, NULL as name, a.body, NULL as bio, NULL as description, a.created_ts, a.fellowship, NULL as avatar_url, NULL as icon_url FROM announcements a
      WHERE a.fellowship == "founders" OR a.fellowship == "all"
      UNION ALL
      SELECT u.id, NULL as title, u.name, NULL as body, u.bio, NULL as description, u.created_ts, u.fellowship, u.avatar_url, NULL as icon_url FROM users u
      WHERE u.fellowship == "founders" OR u.fellowship == "angels"
      UNION ALL
      SELECT p.id, NULL as title, p.name, NULL as body, NULL as bio, p.description, p.created_ts, NULL as fellowship, NULL as avatar_url, p. icon_url FROM projects p
      ORDER BY created_ts DESC
      LIMIT ? OFFSET ?
    `,
    [limitQP, offsetQP]
  )
}

const angelsQuery = (limitQP: number, offsetQP: number) => {
  return db.getAll(`
      SELECT a.id, a.title, NULL as name, a.body, NULL as bio, NULL as description, a.created_ts, a.fellowship, NULL as avatar_url, NULL as icon_url FROM announcements a
      WHERE a.fellowship == "angels" OR a.fellowship == "all"
      UNION ALL
      SELECT u.id, NULL as title, u.name, NULL as body, u.bio, NULL as description, u.created_ts, u.fellowship, u.avatar_url, NULL as icon_url FROM users u
      WHERE u.fellowship == "founders" OR u.fellowship == "angels"
      UNION ALL
      SELECT p.id, NULL as title, p.name, NULL as body, NULL as bio, p.description, p.created_ts, NULL as fellowship, NULL as avatar_url, p. icon_url FROM projects p
      ORDER BY created_ts DESC
      LIMIT ? OFFSET ?
    `,
    [limitQP, offsetQP]
  )
}
