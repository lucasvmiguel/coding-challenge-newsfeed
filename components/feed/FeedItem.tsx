
import { resolveTypes } from '../../graphql/client/types'

import AnnouncementCard, { Announcement } from '../card/AnnouncementCard'
import UserCard, { User } from '../card/UserCard'
import ProjectCard, { Project } from '../card/ProjectCard'

export type Item = (Announcement | User | Project) & { __typename: string, created_ts: string }

type Props = {
  item: Item;
}

export default function FeedItem({ item }: Props) {
  switch (item.__typename) {
    case resolveTypes.Project:
      return <ProjectCard project={item as Project} />
    case resolveTypes.User:
      return <UserCard user={item as User} />
    case resolveTypes.Announcement:
      return <AnnouncementCard announcement={item as Announcement} />
    default:
      return null
  }
}