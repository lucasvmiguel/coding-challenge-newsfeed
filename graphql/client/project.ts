import { gql } from '@apollo/client'

export const PROJECT_QUERY = gql`
  query project($id: Int!) {
    project(id: $id) {
      id
      name
      description
      icon_url
      users {
        id
        name
        avatar_url
      }
    }
  }
`
