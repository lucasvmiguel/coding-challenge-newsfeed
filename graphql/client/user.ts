import { gql } from '@apollo/client'

export const USER_QUERY = gql`
  query user($id: Int!) {
    user(id: $id) {
      id
      name
      bio
      fellowship
      avatar_url
      projects {
        id
        name
        icon_url
      }
    }
  }
`