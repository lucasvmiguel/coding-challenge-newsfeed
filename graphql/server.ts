import { ApolloServer, gql } from 'apollo-server-micro'

import { resolveTypes } from './client/types';
import * as resolvers from './resolvers'


const typeDefs = gql`
  enum Fellowship {
    founders
    angels
    writers
    all
  }

  type Project {
    id: Int!
    name: String!
    description: String!
    icon_url: String!
    users: [User!]!
    created_ts: String!
  }

  type User {
    id: Int!
    name: String!
    bio: String!
    avatar_url: String!
    fellowship: Fellowship!
    projects: [Project!]!
    created_ts: String!
  }

  type Announcement{
    id: Int!
    fellowship: Fellowship!
    title: String!
    body: String!
    created_ts: String!
    updated_ts: String!
  }

  union FeedItem = User | Announcement | Project

  type Query {
    project(id: Int!): Project!
    user(id: Int!): User!
    feed(user_type: Fellowship!, limit: Int, offset: Int): [FeedItem!]!
  }
`;

export const server = new ApolloServer({
  typeDefs,
  resolvers: {
    ...resolvers,
    FeedItem: {
      __resolveType(obj: any, context: any, info: any) {
        if (obj.bio !== null) {
          return resolveTypes.User;
        }

        if (obj.body !== null) {
          return resolveTypes.Announcement;
        }

        if (obj.description !== null) {
          return resolveTypes.Project;
        }

        return null;
      },
    },
  }
})
