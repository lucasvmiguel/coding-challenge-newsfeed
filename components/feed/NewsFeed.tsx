import styled from 'styled-components'

import FeedItem, { Item } from './FeedItem'

type Props = {
  items: Item[];
}

export default function NewsFeed({ items }: Props) {
  return (
    <div>
      <ul>
        {items.map(item => (
          <ListItem key={`${item.id}-${item.__typename}`}>
            <FeedItem item={item} />
          </ListItem>
        ))}
      </ul>
    </div>
  )
}

const ListItem = styled.li`
  list-style-type: none;
  padding-bottom: 20px;
`
