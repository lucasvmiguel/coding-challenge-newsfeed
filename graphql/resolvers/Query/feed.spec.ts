import { Fellowship } from '../../db';
import { FEED_QUERY } from '../../client/feed';
import { server } from '../../server'

const LIMIT_AND_OFFSET = 5;

[
  Fellowship.Founders,
  Fellowship.Writers,
  Fellowship.Angels,
  "invalid",
].forEach(userType => {
  describe(`fetches feed as ${userType}`, () => {
    it('fetches feed', async () => {
      const res = await server.executeOperation({
        query: FEED_QUERY,
        variables: { user_type: userType, limit: LIMIT_AND_OFFSET }
      });
      expect(res).toMatchSnapshot();
    })

    it('fetches feed with offset', async () => {
      const res = await server.executeOperation({
        query: FEED_QUERY,
        variables: { user_type: userType, limit: LIMIT_AND_OFFSET, offset: LIMIT_AND_OFFSET }
      });
      expect(res).toMatchSnapshot();
    })
  })
})