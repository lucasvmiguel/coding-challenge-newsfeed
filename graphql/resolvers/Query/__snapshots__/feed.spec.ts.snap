// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches feed as angels fetches feed 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "__typename": "Announcement",
        "body": "While registration has **closed**, if you've not signed up yet, you're invited to join Build Weekend pitches on Sunday, 2/21! GBW is our flagship cross-fellowship event, bringing together 300+ fellows across 7+ programs for a weekend of tinkering.

If you've already registered, hang tight for more information!",
        "created_ts": "2021-02-14 17:22",
        "fellowship": "all",
        "id": 5,
        "title": "Get ready for Global Build Weekend Pitches next Sunday, 2/21 üåêüõ†!",
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a27.svg",
        "bio": "Helped build ad markets at Appster for many years (NYC adtech, acquired by Verizon). Now applying that optimization / marketplace background to recruiting / FoW tools.

Working on connecting talent to opportunity more efficiently at Talentops.",
        "created_ts": "2021-02-13 17:49",
        "fellowship": "founders",
        "id": 27,
        "name": "Maciej Wilcox",
        "projects": Array [
          Object {
            "icon_url": "https://avatars.dicebear.com/api/jdenticon/a10.svg",
            "id": 10,
            "name": "Blue Onion Labs",
          },
        ],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a30.svg",
        "bio": "Previously: Product / Design leader in B2B SaaS  (ZenRain, Ringomatic, InstaCoupon). Diverse experience across operational products from CRM to full blown accounting in different markets. 

Also interested in future of work, autonomous vehicles, and elder tech. 

I love voice / video and plan on doing something with webrtc.",
        "created_ts": "2021-02-13 16:30",
        "fellowship": "founders",
        "id": 30,
        "name": "Subhaan Madden",
        "projects": Array [],
      },
      Object {
        "__typename": "Project",
        "created_ts": "2021-02-13 14:58",
        "description": "This financial services startup is helping companies make sense of financial transaction data that lives across multiple systems. Their suite of API integrations are aiming to solve a big pain point for accounting teams, serving as a ‚Äúsingle source of truth‚Äù for understanding the full scope of incoming transactions.",
        "icon_url": "https://avatars.dicebear.com/api/jdenticon/a10.svg",
        "id": 10,
        "name": "Blue Onion Labs",
        "users": Array [
          Object {
            "avatar_url": "https://avatars.dicebear.com/api/avataaars/a26.svg",
            "id": 26,
            "name": "Kacy Connelly",
          },
          Object {
            "avatar_url": "https://avatars.dicebear.com/api/avataaars/a27.svg",
            "id": 27,
            "name": "Maciej Wilcox",
          },
        ],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a28.svg",
        "bio": "",
        "created_ts": "2021-02-13 14:08",
        "fellowship": "founders",
        "id": 28,
        "name": "Kourtney Torres",
        "projects": Array [],
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches feed as angels fetches feed with offset 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a29.svg",
        "bio": "Portuguese with an electronics background turned computer scientist turned startup guy. Former CTO of EdTech startup Thinkpad. Used to live in SF, just moved to LA after 5 years in NY.",
        "created_ts": "2021-02-13 11:15",
        "fellowship": "founders",
        "id": 29,
        "name": "Cristiano Molloy",
        "projects": Array [],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a26.svg",
        "bio": "",
        "created_ts": "2021-02-12 15:57",
        "fellowship": "founders",
        "id": 26,
        "name": "Kacy Connelly",
        "projects": Array [
          Object {
            "icon_url": "https://avatars.dicebear.com/api/jdenticon/a10.svg",
            "id": 10,
            "name": "Blue Onion Labs",
          },
        ],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a25.svg",
        "bio": "I recently spent 3 years on the investment team at Andreessen Horowitz focused on the Consumer sector. 

I'm interested in companies that are enabling individuals to make a living from their unique audiences.

I am an active angel investor! HMU if you want: fundraising advice, product/strategy jams...",
        "created_ts": "2021-02-10 12:29",
        "fellowship": "angels",
        "id": 25,
        "name": "Ruby-Mae Middleton",
        "projects": Array [],
      },
      Object {
        "__typename": "Project",
        "created_ts": "2021-02-08 15:30",
        "description": "A software company that provides a fintech API allowing other platforms to integrate investment accounts into their product, easily. Many modern fintech services allow you to hold cash, like Venmo and Apple Pay. That money could be, instead, invested. Atomic has found some initial traction, with a few companies signed up that give it around $300M in AUM, providing the startup with $1.5M in ARR. After every fintech added checking accounts, perhaps investments accounts are the next step.",
        "icon_url": "https://avatars.dicebear.com/api/jdenticon/a9.svg",
        "id": 9,
        "name": "Atomic",
        "users": Array [
          Object {
            "avatar_url": "https://avatars.dicebear.com/api/avataaars/a24.svg",
            "id": 24,
            "name": "Raees Carver",
          },
        ],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a24.svg",
        "bio": "I think building is fun. Thanks to no-code, I've built over a dozen projects in 2020. I'm on the lookout for a cofounder to build something amazing (see interests below). In the meanwhile, I'm advising founders on how to get from 0-1 with no-code approach.",
        "created_ts": "2021-02-05 14:04",
        "fellowship": "founders",
        "id": 24,
        "name": "Raees Carver",
        "projects": Array [
          Object {
            "icon_url": "https://avatars.dicebear.com/api/jdenticon/a9.svg",
            "id": 9,
            "name": "Atomic",
          },
        ],
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches feed as founders fetches feed 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "__typename": "Announcement",
        "body": "While registration has **closed**, if you've not signed up yet, you're invited to join Build Weekend pitches on Sunday, 2/21! GBW is our flagship cross-fellowship event, bringing together 300+ fellows across 7+ programs for a weekend of tinkering.

If you've already registered, hang tight for more information!",
        "created_ts": "2021-02-14 17:22",
        "fellowship": "all",
        "id": 5,
        "title": "Get ready for Global Build Weekend Pitches next Sunday, 2/21 üåêüõ†!",
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a27.svg",
        "bio": "Helped build ad markets at Appster for many years (NYC adtech, acquired by Verizon). Now applying that optimization / marketplace background to recruiting / FoW tools.

Working on connecting talent to opportunity more efficiently at Talentops.",
        "created_ts": "2021-02-13 17:49",
        "fellowship": "founders",
        "id": 27,
        "name": "Maciej Wilcox",
        "projects": Array [
          Object {
            "icon_url": "https://avatars.dicebear.com/api/jdenticon/a10.svg",
            "id": 10,
            "name": "Blue Onion Labs",
          },
        ],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a30.svg",
        "bio": "Previously: Product / Design leader in B2B SaaS  (ZenRain, Ringomatic, InstaCoupon). Diverse experience across operational products from CRM to full blown accounting in different markets. 

Also interested in future of work, autonomous vehicles, and elder tech. 

I love voice / video and plan on doing something with webrtc.",
        "created_ts": "2021-02-13 16:30",
        "fellowship": "founders",
        "id": 30,
        "name": "Subhaan Madden",
        "projects": Array [],
      },
      Object {
        "__typename": "Project",
        "created_ts": "2021-02-13 14:58",
        "description": "This financial services startup is helping companies make sense of financial transaction data that lives across multiple systems. Their suite of API integrations are aiming to solve a big pain point for accounting teams, serving as a ‚Äúsingle source of truth‚Äù for understanding the full scope of incoming transactions.",
        "icon_url": "https://avatars.dicebear.com/api/jdenticon/a10.svg",
        "id": 10,
        "name": "Blue Onion Labs",
        "users": Array [
          Object {
            "avatar_url": "https://avatars.dicebear.com/api/avataaars/a26.svg",
            "id": 26,
            "name": "Kacy Connelly",
          },
          Object {
            "avatar_url": "https://avatars.dicebear.com/api/avataaars/a27.svg",
            "id": 27,
            "name": "Maciej Wilcox",
          },
        ],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a28.svg",
        "bio": "",
        "created_ts": "2021-02-13 14:08",
        "fellowship": "founders",
        "id": 28,
        "name": "Kourtney Torres",
        "projects": Array [],
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches feed as founders fetches feed with offset 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a29.svg",
        "bio": "Portuguese with an electronics background turned computer scientist turned startup guy. Former CTO of EdTech startup Thinkpad. Used to live in SF, just moved to LA after 5 years in NY.",
        "created_ts": "2021-02-13 11:15",
        "fellowship": "founders",
        "id": 29,
        "name": "Cristiano Molloy",
        "projects": Array [],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a26.svg",
        "bio": "",
        "created_ts": "2021-02-12 15:57",
        "fellowship": "founders",
        "id": 26,
        "name": "Kacy Connelly",
        "projects": Array [
          Object {
            "icon_url": "https://avatars.dicebear.com/api/jdenticon/a10.svg",
            "id": 10,
            "name": "Blue Onion Labs",
          },
        ],
      },
      Object {
        "__typename": "Announcement",
        "body": "So you've just raised or are about to...what next? Join Nate to learn about the things most people don't tell you in setting up your business: insurance, HR, payroll, accounting, PR, IT, benefits, and much more.",
        "created_ts": "2021-02-11 15:45",
        "fellowship": "founders",
        "id": 4,
        "title": "Workshop: Understanding the back-office necessities around a fundraise",
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a25.svg",
        "bio": "I recently spent 3 years on the investment team at Andreessen Horowitz focused on the Consumer sector. 

I'm interested in companies that are enabling individuals to make a living from their unique audiences.

I am an active angel investor! HMU if you want: fundraising advice, product/strategy jams...",
        "created_ts": "2021-02-10 12:29",
        "fellowship": "angels",
        "id": 25,
        "name": "Ruby-Mae Middleton",
        "projects": Array [],
      },
      Object {
        "__typename": "Project",
        "created_ts": "2021-02-08 15:30",
        "description": "A software company that provides a fintech API allowing other platforms to integrate investment accounts into their product, easily. Many modern fintech services allow you to hold cash, like Venmo and Apple Pay. That money could be, instead, invested. Atomic has found some initial traction, with a few companies signed up that give it around $300M in AUM, providing the startup with $1.5M in ARR. After every fintech added checking accounts, perhaps investments accounts are the next step.",
        "icon_url": "https://avatars.dicebear.com/api/jdenticon/a9.svg",
        "id": 9,
        "name": "Atomic",
        "users": Array [
          Object {
            "avatar_url": "https://avatars.dicebear.com/api/avataaars/a24.svg",
            "id": 24,
            "name": "Raees Carver",
          },
        ],
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches feed as invalid fetches feed 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [UserInputError: Variable "$user_type" got invalid value "invalid"; Value "invalid" does not exist in "Fellowship" enum.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches feed as invalid fetches feed with offset 1`] = `
Object {
  "data": undefined,
  "errors": Array [
    [UserInputError: Variable "$user_type" got invalid value "invalid"; Value "invalid" does not exist in "Fellowship" enum.],
  ],
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches feed as writers fetches feed 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "__typename": "Announcement",
        "body": "While registration has **closed**, if you've not signed up yet, you're invited to join Build Weekend pitches on Sunday, 2/21! GBW is our flagship cross-fellowship event, bringing together 300+ fellows across 7+ programs for a weekend of tinkering.

If you've already registered, hang tight for more information!",
        "created_ts": "2021-02-14 17:22",
        "fellowship": "all",
        "id": 5,
        "title": "Get ready for Global Build Weekend Pitches next Sunday, 2/21 üåêüõ†!",
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a22.svg",
        "bio": "Creator & Writer at Urbane.",
        "created_ts": "2021-01-29 10:39",
        "fellowship": "writers",
        "id": 22,
        "name": "Harvie Drummond",
        "projects": Array [],
      },
      Object {
        "__typename": "Announcement",
        "body": "On Deck Build Weekend across ALL PROGRAMS and ALL COHORTS. Unlike other On Deck events, you must sign up to participate in Build Weekend. Please note: you do not need to have a project idea or a team to sign up!",
        "created_ts": "2021-01-26 16:54",
        "fellowship": "all",
        "id": 3,
        "title": "Announcing: February 2021 Global Build Weekend!",
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a20.svg",
        "bio": "Currently working at IBM as a Data & AI Sales Engineer. Freelance copywriting for a handful of interesting companies. Interested in AI/ML, Bitcoin, digital content, investing, and startups.",
        "created_ts": "2021-01-22 19:30",
        "fellowship": "writers",
        "id": 20,
        "name": "Selina Peel",
        "projects": Array [],
      },
      Object {
        "__typename": "Announcement",
        "body": "What is Throckmorton Day, you ask? All will be revealed in time‚Äîwe‚Äôre intentionally keeping it vague for now‚Äîbut high level, it‚Äôll be a wild day-long write-a-thon with ‚Äúcelebrity‚Äù judges, fabulous prizes, and hijinks galore, open to current Fellows and alums.

Save the date if you can!",
        "created_ts": "2021-01-13 11:49",
        "fellowship": "writers",
        "id": 2,
        "title": "Semi-Annual Throckmorton Day",
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`fetches feed as writers fetches feed with offset 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a14.svg",
        "bio": "I write a long-form newsletter that covers hardware engineering, science, my thoughts on current events, the advice I am applying, and experiments I am running.

HW engineer at Apple",
        "created_ts": "2021-01-10 12:45",
        "fellowship": "writers",
        "id": 14,
        "name": "Devon Mccoy",
        "projects": Array [],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a8.svg",
        "bio": "writer, strategist, advisor, MBA, father, culinary arts enthusiast",
        "created_ts": "2020-12-24 16:24",
        "fellowship": "writers",
        "id": 8,
        "name": "Bret York",
        "projects": Array [],
      },
      Object {
        "__typename": "Announcement",
        "body": "Amidst this crazy 2020, this community has been a beacon of hope that there is a bright future ahead. We‚Äôre so grateful you made the choice to join this community.

We hope you all are able to spend some time with loved ones ‚Äì either virtually, or in-person if safe ‚Äì and enjoy a few days off.",
        "created_ts": "2020-12-24 13:02",
        "fellowship": "all",
        "id": 1,
        "title": "Happy Holidays, from the On Deck Team",
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a9.svg",
        "bio": "Hi there! I‚Äôm currently work in brand building @ Facebook. I have a background in finance and community building for startups I write about culture, social norms, and psychology frameworks.",
        "created_ts": "2020-12-24 11:53",
        "fellowship": "writers",
        "id": 9,
        "name": "Zahraa Broadhurst",
        "projects": Array [],
      },
      Object {
        "__typename": "User",
        "avatar_url": "https://avatars.dicebear.com/api/avataaars/a5.svg",
        "bio": "I run a consultancy business, which provides digital marketing and AI services. I've worked with startups and organizations across the world including Germany and Netherlands. Culture and society are my favourite topics to read and write about. I have always been interested in writing about human behaviours and what my experiences have taught me about life in general. At this point in my writing journey, I am most interested in learning how to make writing a lifelong career.",
        "created_ts": "2020-12-17 19:13",
        "fellowship": "writers",
        "id": 5,
        "name": "Mayur Quintana",
        "projects": Array [],
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
